backend/internal/system/crypto/encrypt/service.go

```diff
@@ -66,18 +67,26 @@ func GetEncryptionService() *EncryptionService {
 	return instance
 }
 
-// initEncryptionService initializes the EncryptionService from configuration sources.
+// initCryptoService initializes the CryptoService from configuration sources.
 func initEncryptionService() (*EncryptionService, error) {
 	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, "EncryptionService"))
-	// Try to get key from the application configuration
-	config := config.GetThunderRuntime().Config.Crypto.Key // Use the correct config getter
-
-	// Check if crypto configuration exists
-	if config != "" {
-		key, err := hex.DecodeString(config)
-		if err == nil {
-			logger.Debug("Using crypto key from configuration")
-			return NewEncryptionService(key)
+	cryptoFilePath := config.GetThunderRuntime().Config.Security.CryptoFile
+
+	if cryptoFilePath != "" {
+		var keyBytes []byte
+		var err error
+		//nolint:gosec // G304 ignored because: File path is provided only via config file
+		fileData, fileErr := os.ReadFile(cryptoFilePath)
+		if fileErr == nil {
+			// File read successfully, so decode its content
+			logger.Debug("Attempting to load crypto key from file", log.String("path", cryptoFilePath))
+			keyBytes, err = hex.DecodeString(string(fileData))
+			if err == nil {
+				logger.Info("Successfully loaded crypto key from file")
+				return NewEncryptionService(keyBytes)
+			}
+			logger.Warn("Failed to decode hex key from file, trying as literal string",
```

Comment:
This should be an error. We should fail the server startup if the crypto key is not found/ decode failed.

Shall we also remove the existing key generation logic at runtime?
@hwupathum this is the expected behaviour right?